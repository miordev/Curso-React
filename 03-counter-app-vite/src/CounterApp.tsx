import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ({ value }: any) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
    // setCounter((c) => c + 1);
  };

  const handleSubtract = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2 data-testid="test-counter">{counter}</h2>
      <button aria-label="increase-btn" onClick={handleAdd}>
        +1
      </button>
      <button aria-label="decrease-btn" onClick={handleSubtract}>
        -1
      </button>
      <button aria-label="reset-btn" onClick={handleReset}>
        Reset
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
