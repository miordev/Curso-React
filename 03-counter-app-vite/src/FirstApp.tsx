import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle }: any) => {
  return (
    <>
      <h3>{title}</h3>
      <p>{subtitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired,
};
