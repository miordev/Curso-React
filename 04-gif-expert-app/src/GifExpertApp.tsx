import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categoryList, setCategoryList] = useState(['One Punch']);

  const onAddCategory = (newCategory: string): void => {
    if (!categoryList.includes(newCategory)) {
      setCategoryList([newCategory, ...categoryList]);
    }
  };

  return (
    <>
      <h1>GifExpertApp 2</h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categoryList.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
