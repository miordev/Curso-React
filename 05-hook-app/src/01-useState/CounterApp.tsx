import { useState } from 'react';
export const CounterApp = () => {
  const [state, setState] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
  });

  const { counter1, counter2, counter3 } = state;

  return (
    <>
      <div>Counter1: {counter1}</div>
      <div>Counter2: {counter2}</div>
      <div>Counter3: {counter3}</div>
      <hr />
      <button
        className="btn"
        onClick={() =>
          setState({
            ...state,
            counter1: counter1 + 1,
          })
        }
      >
        +1
      </button>
    </>
  );
};
