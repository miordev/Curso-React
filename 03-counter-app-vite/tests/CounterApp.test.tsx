import { render, screen, fireEvent } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('Tests in CounterApp', () => {
  const initialValue = 100;

  test('should do a match with the snapshot', () => {
    const { container } = render(<CounterApp value={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  test('should show the initial value', () => {
    render(<CounterApp value={initialValue} />);
    expect(screen.getByTestId('test-counter').innerHTML).toBe(initialValue.toString());
  });

  test('should increase the value', () => {
    render(<CounterApp value={initialValue} />);
    const increaseBtn = screen.getByRole('button', { name: 'increase-btn' });
    fireEvent.click(increaseBtn);
    expect(screen.getByTestId('test-counter').innerHTML).toBe((initialValue + 1).toString());
  });

  test('should decrease the value', () => {
    render(<CounterApp value={initialValue} />);
    const decreaseBtn = screen.getByRole('button', { name: 'decrease-btn' });
    fireEvent.click(decreaseBtn);
    expect(screen.getByTestId('test-counter').innerHTML).toBe((initialValue - 1).toString());
  });

  test('should reset the value', () => {
    render(<CounterApp value={initialValue} />);
    const increaseBtn = screen.getByRole('button', { name: 'increase-btn' });
    const resetBtn = screen.getByRole('button', { name: 'reset-btn' });

    fireEvent.click(increaseBtn);
    fireEvent.click(increaseBtn);
    fireEvent.click(increaseBtn);
    fireEvent.click(resetBtn);

    expect(screen.getByTestId('test-counter').innerHTML).toBe(initialValue.toString());
  });
});
