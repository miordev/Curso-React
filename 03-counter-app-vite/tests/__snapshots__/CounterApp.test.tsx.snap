// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in CounterApp should do a match with the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    data-testid="test-counter"
  >
    100
  </h2>
  <button
    aria-label="increase-btn"
  >
    +1
  </button>
  <button
    aria-label="decrease-btn"
  >
    -1
  </button>
  <button
    aria-label="reset-btn"
  >
    Reset
  </button>
</div>
`;
